### ISSUE TITLE
reconsider first calculate(model) in my_initializeModel$run in MCMC

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#625]
calculate(model) in this context fails when a dynamic index is not initialized.

Per @danielturek comment by email:
That first calculate call I believe is only necessary to propagate things downwards *from* RHS-only nodes (truly, the top level nodes) downwards through any "top level deterministic" nodes, and finally *into* the top-level stochastic nodes.  If that truly is the only use, then yes, the full calculate(model) call could be relaxed -- but I'll admit I could be forgetting some other reason.  But looking at the code, that seems to be the case.

Here's a possibility to consider. Happy to discuss here or perhaps even better in our next meeting.

1) call calculate only on topOnly=TRUE, determOnly=TRUE nodes. 
2) then initiate the current loop in my_initializeModel that loops over the stoch nodes and samples from their priors and then calcs and deps of the stoch node

alternatively we could find the RHSonly nodes and call calculate on their dependencies. I don't think we have any ability currently get only the RHSonly nodes.

**Comment by danielturek** [comment]
Ideas seem reasonable, just one correction, we do have the ability to get RHSonly nodes, take a look at current `initializeModel`, that's a core part of what it does.

**Comment by paciorek** [comment]
ah... It's not user-facing so I didn't realize we had that in the getMaps(). So looks like the (simple) change to the initialization would be to replace calculate(model) with:
```
calculate(model, model$getDependencies(model$getMaps('nodeNamesRHSonly'), determOnly = TRUE)
```

**Comment by danielturek** [comment]
Sounds reasonable.  Let's discuss on Wednesday.

**Comment by paciorek** [comment]
addressed by PR #627
